components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/self-registration-process.jsp

```diff
@@ -174,8 +175,13 @@
                 selfUserRegistrationRequest.setUser(selfRegistrationUser);
                 selfUserRegistrationRequest.setProperties(properties);
 
+                Map<String, String> requestHeaders = new HashedMap();
```

Comment:
We should move the java logic to a Java code. It is not good to have so much Java logic within JSP.